frappe/utils/global_search.py

```diff
@@ -418,10 +418,19 @@ def search(text, start=0, limit=20, doctype=""):
 	from frappe.desk.doctype.global_search_settings.global_search_settings import get_doctypes_for_global_search
 
 	results = []
-	texts = [t.strip() for t in text.split('&') if t]
-	priorities = get_doctypes_for_global_search()
-	allowed_doctypes = ",".join(["'{0}'".format(dt) for dt in priorities])
-	for text in texts:
+	sorted_results = []
+
+	priorities = frappe.cache().hget("global_search", "search_priorities")
```

Comment:
Why are you using data directly from cache... since it looks like cache will be built only on global search settings update... it might not work for new users with default settings...

Reply:
- Global Search is not User Specific, so settings it as cache saves some time.
- Also if priorities is not set, then on L427 it'll be set.